### DISCUSSION TITLE
How to create a one chromosome cool matrix

### DISCUSSION THREAD

**Discussion by LAlvarezGlez** [open2c/cooler#370] [Q&A] [answered]
Hi,

I am quite new with the cool format and I am not able to understand how to do this. I have a genome-wide .cool matrix and to perform some analysis I want just a chromosome X .cool matrix. How do I do that?

Many thanks

Lucía

**Comment by nicholas-moskwa** [comment]
I would also like an answer to lucia's question

**Comment by Phlya** [comment]
Apologies for not replying to the original question. I am wondering about the reason you need this? You can easily extract just one chromosome from a whole-genome cooler file for analysis.

Otherwise, @nvictus would know the best way to do this... It should be straightforward, you just need pixels and bins for the right chromosome, and then use `cooler.create_cooler()` in python API, or `cooler load` in CLI.

**Comment by nicholas-moskwa** [comment]
Hi @Phlya , well I am trying to analyze my data using FIREcaller: https://yunliweb.its.unc.edu/FIREcaller/download.php
This Rscript states that it can use .cool files but I haven't had any luck. Their tutorial actually uses a list of chromosome matrix files, so ultimately I think I am going to need to turn the .cool files into an uncompressed location to signal matrix.

**Comment by nvictus** [comment, marked as answer]
Looks to me like there's a [missing function implementation](https://github.com/yycunc/FIREcaller/blob/master/R/FIREcaller.R#L292) in the FIREcaller package: `cooler2bedpe`.

If you decide to try hacking the script to use a dataframe or matrix instead of a file, we recently discovered that you can very easily run the Python cooler package from within R using [reticulate](https://rstudio.github.io/reticulate/) and extract what you need. In case it helps:

```r
reticulate::use_condaenv("myenv")  # a conda environment with cooler installed
cooler <- reticulate::import("cooler")

clr <- cooler$Cooler("path/to/file.cool::resolutions/10000")
mat <- clr$matrix(balance = FALSE)$fetch("chr1")  # raw counts, unnormalized, as array
dat <- clr$matrix(balance = FALSE, as_pixels = TRUE)$fetch("chr1")   # raw counts, unnormalized, as dataframe

```

**Comment by nicholas-moskwa** [comment]
You know @nvictus , you are so helpful and I want you to know I really appreciate the assistance. Your R code is generating the matrix I think I need for my analysis! One more question, I don't mind using R for sub-setting the .cool file, but does cooler have a CLI command for doing this?

**Comment by nvictus** [comment]
It sure does! [`cooler dump`](https://cooler.readthedocs.io/en/latest/cli.html#cooler-dump) is what you are looking for. See the `--range` option.
